### ISSUE TITLE
When stopping playback from the detail view, show the detail view again on stop

### ISSUE THREAD

**Issue by softworkz** [EmbySupport/Emby.WMC.UI#37]
Requested here: https://emby.media/community/index.php?/topic/123545-ask-for-deletion-when-stop-viewing-a-tv-recording/

**Comment by ebr11** [comment]
There is an open FR for the whole system for this.

https://emby.media/community/index.php?/topic/66113-offer-to-delete-show-after-watching/&do=findComment&comment=657135

**Comment by softworkz** [comment]
Thanks for the reference. For clarification - this is about showing this view when stopping or playback ended:

![image](https://github.com/EmbySupport/Emby.WMC.UI/assets/4985349/0e9cf485-f7dc-40b1-98a2-99ac544b68aa)

**Comment by softworkz** [comment]
WMC doesn't have the ability to "play a series", neither do we (for recorded tv). And as far as I've understood, this would be primarily for recorded tv.

But let's look at the case of regular series libraries.

You would start from here:

![image](https://github.com/EmbySupport/Emby.WMC.UI/assets/4985349/35a450b6-5ad6-463c-b746-32d9a848795e)

and you would also be taken back to the screen above.

There's no immediate delete button. To delete though, you can tap right and a few times down until you get to the Delete action. When you tap OK, you will get this:

![image](https://github.com/EmbySupport/Emby.WMC.UI/assets/4985349/66f8998e-833c-4f36-9763-06e8c6132e5c)

So that procedure would appear to be safe enough, I'd say, wouldn't you?

**Comment by ebr11** [comment]
> And as far as I've understood, this would be primarily for recorded tv.

There is not necessarily a distinction.  It is possible to deposit recordings directly into a "normal" Emby library.  I don't see why we would try to make a distinction on how this would work either but, if we were to implement the feature requested in the topic I linked, I would think it would be done per-library and on an opt-in basis.

> So that procedure would appear to be safe enough, I'd say, wouldn't you?

Nope.  That right there is one of the major issues - if the user did what you just described, they just deleted the entire series, not the episode they just watched.

> An alternative would be to show the details for the episode which was just playing.

Perhaps and, that is actually what the Android TV app does now.  However, that still doesn't get to the real FR.  Think about the "workflow".  If one is watching several episodes in a row, this approach will only offer to delete the last one.

Therefore, some sort of optional dialog actually at the end of playback is what would be required to get the feature how they probably really want it.

**Comment by softworkz** [comment]
> > So that procedure would appear to be safe enough, I'd say, wouldn't you?
> 
> Nope. That right there is one of the major issues - if the user did what you just described, they just deleted the entire series, not the episode they just watched.

The message text could be more clear, stating that it affects the entire series. 
Anyway, it's up to the admin to decide whether to allow this or not. And as long as it's not a single-person setup, an admin would probably not allow deletions anyway.
But if they allow this, then users can always get to that view with the "Delete" action. The only point in this context (WMC UI) is whether to show it that item-view on stop or not.

> > An alternative would be to show the details for the episode which was just playing.
> 
> Perhaps and, that is actually what the Android TV app does now

Yes, it makes sense.

> Perhaps and, that is actually what the Android TV app does now. However, that still doesn't get to the real FR. Think about the "workflow". If one is watching several episodes in a row, this approach will only offer to delete the last one.

For that part, I'm rather on the side of what you wrote in the FR topic, that user interest in this is not hitting a reasonable bar to even think about it.

If we take any effort in that direction, then I think that a disc-space management like WMC has would make much more sense than a manual deletion popup after playback.

**Comment by softworkz** [comment]
> > And as far as I've understood, this would be primarily for recorded tv.
> 
> There is not necessarily a distinction. It is possible to deposit recordings directly into a "normal" Emby library. I don't see why we would try to make a distinction on how this would work either but, 

The distinction would be from which place the item would have been accesses (recorded tv or library), but maybe not a great solution.

> if we were to implement the feature requested in the topic I linked, I would think it would be done per-library and on an opt-in basis.

But to what would one opt-in in that case? 

- Being able to do deletions in general 
  (isn't this covered by user permissions)
- Showing the item-view on playback-stop (or not)
- Showing or not showing the delete button in the item-view on playback-stop
  (but always showing the delete-button when it's not an 'after-playback' situation)

**Comment by softworkz** [comment]
> The opt-in would be whether or not to show some ability to delete the item directly after playback - each playback. Like I said in the forum, this could be tied in with the "next episode" popup that is already there.
> 
> However - two major caveats:
> 
> I agree I don't think there are enough users wanting this to warrant the effort

Right, and I also think that there are more reasonable and more generally useful things we could do in this area:

- Like mentioned above, introduce a "Keep until" setting 
  which the primary purpose of being able to set "Keep forever" given we would have an "auto-delete" task for cleaning up old TV recordings alongside additional options (per-item and per-series)
- Another idea would be that non-admins can only ever execute a "schedule for deletion" action
  Nothing would be deleted instantly in this case, and the admin would have choices how to deal with "scheduled deletions":
  - Put them into an approval queue for the admin to approve
  - Delete them only after a certain period of time (admin can review and cancel)
  - Delete them only when storage gets low



> Therefore, if you want to do anything in the theme, I would just present the detail screen of the last played item at the end of playback.

Yup, that's the only action I'm intending to take here.
